### ISSUE TITLE
Opening neotree from dedicated window messes window configuration

### ISSUE THREAD

**Issue by bmag** [jaypei/emacs-neotree#124]
To reproduce:
1. open emacs: `emacs -Q -l /path/to/neotree.el`
2. open scratch buffer and make the window dedicated: `M-: (set-window-dedicated nil t)`
3. open neotree: `M-x neotree`. The windows are messed up.
Expected window configuration:
```
+---------+--------------------------+
|         |                          |
|         |                          |
|         |                          |
| neotree |      scratch             |
|         |                          |
|         |                          |
|         |                          |
+---------+--------------------------+
```

Actual window configuration:
```
+--------+--------+--------------+---------+
|        |        |              |         |
|        |        |              |         |
|        |        |              |         |
|scratch |scratch |   scratch    | neotree |
|        |        |              |         |
|        |        |              |         |
|        |        |              |         |
|        |        |              |         |
+--------+--------+--------------+---------+
```

Note that setting `neo-modern-sidebar` to `t` solves this issue.

I suspect [these lines](https://github.com/jaypei/emacs-neotree/blob/dev/neotree.el#L635-L637) are the problem:
```elisp
(progn
  (select-window (neo-global--get-first-window))
  (split-window-horizontally)))
```
Replacing them with something like this should do the trick (didn't test it):
```elisp
(select-window (split-window (neo-global--get-first-window) nil 'left))
```

**Comment by jaypei** [comment]
Hi bmag,

I think it should remove the old method to split window, and set 'modern-sidebar' to be default.
But if the only one window is a dedicated window, find-file-other-window will be failed.
Please tell me in what situations you can set the only one dedicated-window except NeoTree?

**Comment by bmag** [comment]
> if the only one window is a dedicated window, find-file-other-window will be failed

Are you sure about it? Won't `find-file-other-window` create a new window? The behavior also depends on frame size and split thresholds (`split-width-threshold`, `split-height-threshold`).

> in what situations you can set the only one dedicated-window except NeoTree?

There can be all sorts of situations where a user will manually dedicate a window, usually for users that wish to customize how windows are displayed. For an example, you look at this [SE question](emacs.stackexchange.com/questions/2189/how-can-i-prevent-a-command-from-using-specific-windows), where a custom `toggle-window-dedicated` command is used to prevent `next-error` from hiding the `*grep*` buffer.
In general, the old ("non-modern") way of showing NeoTree created a new window (as a split of an existing window), but displayed NeoTree in the old window and the original buffer in the new window. That meant that NeoTree's window inherited the window parameters of the original buffer's old window, and the original buffer's new window had none of the old window parameters. So any behavior that relies on window parameters would change (`quit-window` is one example).
It's probably rare to have a use-case for users to be affected by this bug (hence it wasn't reported before), but I played quite a bit with window management in emacs so I encountered this bug.
